lib/services/database.go

```diff
@@ -239,6 +240,63 @@ func NewDatabaseFromRedshiftCluster(cluster *redshift.Cluster) (types.Database,
 	})
 }
 
+// NewDatabaseFromElastiCacheConfigurationEndpoint creates a database resource
+// from ElastiCache configuration endpoint.
+func NewDatabaseFromElastiCacheConfigurationEndpoint(cluster *elasticache.ReplicationGroup, extraLabels map[string]string) (types.Database, error) {
+	if cluster.ConfigurationEndpoint == nil {
+		return nil, trace.BadParameter("missing configuration endpoint")
+	}
+
+	return newElastiCacheDatabase(cluster, cluster.ConfigurationEndpoint, awsutils.ElastiCacheConfigurationEndpoint, extraLabels)
+}
+
+// NewDatabasesFromElastiCacheNodeGroups creates database resources from
+// ElastiCache node groups.
+func NewDatabasesFromElastiCacheNodeGroups(cluster *elasticache.ReplicationGroup, extraLabels map[string]string) (types.Databases, error) {
+	var databases types.Databases
+	for _, nodeGroup := range cluster.NodeGroups {
+		if nodeGroup.PrimaryEndpoint != nil {
+			database, err := newElastiCacheDatabase(cluster, nodeGroup.PrimaryEndpoint, awsutils.ElastiCachePrimaryEndpoint, extraLabels)
```

Comment:
Looks like endpoint and endpoint type are tightly coupled and passing reader/primary combination is always wrong. Is there any reason why we need both of them?